samples/net/mdns_responder/src/service.c

```diff
@@ -19,11 +19,19 @@ LOG_MODULE_REGISTER(mdns_echo_service, LOG_LEVEL_DBG);
 /* A default port of 0 causes bind(2) to request an ephemeral port */
 #define DEFAULT_PORT 0
 
-static void welcome(int fd)
+static int welcome(int fd)
 {
 	static const char msg[] = "Bonjour, Zephyr world!\n";
 
-	send(fd, msg, sizeof(msg), 0);
+	int r = send(fd, msg, sizeof(msg), 0);
+
+	if (r == -1) {
+		NET_DBG("send() failed (%d)", errno);
+		close(fd);
```

Comment:
This certainly works, but it would look uniform if the `close()` is done in `service()` so same way as other functions return -1 and then we call close.

Reply:
I 100% agree with you. Good idea!